File: backend/src/main/java/com/edusmart/assignment/model/Assignment.java
package com.edusmart.assignment.model;

import com.edusmart.user.model.Teacher;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "assignment")
@Getter
@Setter
public class Assignment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int assignment_id;
    private String title;
    private String description;
    private LocalDateTime start_date;
    private LocalDateTime due_date;
    private float max_score;
    private String filepath;
    @ManyToOne
    @JoinColumn(name = "teacher_id")
    private Teacher teacher;
}

File: frontend/types/assignment.ts
import { SubmissionHistory } from "./submission";

export interface AssignmentInCourse {
    assignment_id: Number,
    title: string,
    description: string,
    start_date: string,
    due_date: string,
    max_score: Number,
    filepath: String,
    submission: SubmissionHistory
}